crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -351,23 +353,122 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
+            }
+        }
+        /// Gets the output.
+        pub const fn get_txout(&self) -> &bitcoin::TxOut {
+            &self.txout
+        }
+        /// Gets the height that the output was commited.
+        pub const fn get_height(&self) -> u32 {
+            self.height
+        }
+        pub const fn get_time(&self) -> u32 {
+            self.time
+        }
+    }
+    /// A [`HashMap<bitcoin::OutPoint, UtxoData>`] alias.
+    ///
+    /// Use HashMap to store and fast-find all needed UTXOs related data.
+    pub type UtxoMap = HashMap<bitcoin::OutPoint, UtxoData>;
+}
+/// Module to delegate local-time and Median Time Past functions.
+///
+/// The consumer of `Floresta-chain` has the option to implement it or just pass any (chaintime, mtp) as (u32,u32)
+///
+/// # Examples
+/// ```
+/// use chain_time::Chaintime;
+///
+/// fn check_chain_time<C: Chaintime>(chain_time_obj: C) -> bool {
+///     if !(let u32 = chain_time_obj.get_time()){
+///         return false;
+///     }
+///     if !(let u32 = chain_time_obj.get_mtp()){
+///         return false;
+///     }
+///     true
+/// }
+///
+/// //(u32,u32) already have a impl inside chain_time module. See [`chain_time::SimpleChainTime`]
+///
+/// let zero_unixtimestamp: u32 = 0; // zero values will disable time checks
+///
+/// let zero_mtp: u32 = 0; // zero values will disable time checks
+///
+/// let entry: SimpleChainTime = (zero_unixtimestamp, zero_mtp); // Cast a (u32,u32) as [`chain_time::SimpleChainTime`]
+///
+/// assert(check_chain_time(entry)); // Using the provided type
+/// assert(check_chain_time((zero_unixtimestamp, zero_mtp)); // raw tuple
+/// ```
+pub mod chain_time {
+
+    /// (u32,u32) type alias that already implements chaintime
+    pub type SimpleChainTime = (u32, u32);
+    /// Disable Constant for [`chain_time::SimpleChainTime`]
+    ///
+    /// Meant to be used in cases to disable time verifications.
+    pub const DISABLE_TIME: SimpleChainTime = (0, 0);
+    /// Trait to return time-related context of the chain.
+    ///
+    /// [`get_time()`] should return a the latest [unix timestamp](https://en.wikipedia.org/wiki/Unix_time) when the consumer has time-notion.
+    ///
+    /// [`get_mtp()`] should return the [Median Time Past](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki) if the consumer has control of it.
+    ///
+    /// if the consumer does not have any time notion or MTP control, its safe to use [`(0u32,0u32)`] to disable any validations on time.
+    pub trait Chaintime {
+        /// Should return a unix timestamp or 0 to skip any time related validation.    
+        fn get_time(&self) -> u32;
+
+        /// Should return a BlockTimestamp or 0 to skip any mtp related validation.   
+        fn get_mtp(&self) -> u32;
+    }
+    impl Chaintime for SimpleChainTime {
+        fn get_time(&self) -> u32 {
+            self.0.clone()
+        }
+        fn get_mtp(&self) -> u32 {
+            self.1.clone()
+        }
+    }
+}
```

Comment:
Now Chaintime is a trit to simple return the mtp and a unix timestamp

Reply:
How will this struct see the mtp? This is a computed information, not something we get from the environment like wall time.